src/foam/nanos/ruler/predicate/PropertyChangePredicate.js

```diff
@@ -0,0 +1,34 @@
+foam.CLASS({
+  package: 'foam.nanos.ruler.predicate',
+  name: 'PropertyChangePredicate',
+
+  documentation: 'A predicate that returns true when a specific property has changed. This predicate expects to be able to pull the old and new object out of the context to do this check.',
+
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+
+  javaImports: [
+    'foam.core.FObject',
+    'static foam.mlang.MLang.*'
+  ],
+  properties: [
+    {
+      class: 'String',
+      name: 'propName'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        FObject nu  = (FObject) NEW_OBJ.f(obj);
+        FObject old = (FObject) OLD_OBJ.f(obj);
+        return old != null && nu != null &&
+          ! foam.util.SafetyUtil.equals(
```

Comment:
`PropertyInfo.diff()` would be more appropriate since we're diffing two properties, wouldn't it?

Reply:
SafetyUtil is more convient here because it returns a boolean directly. Is there a functional difference between the two? I can't find proper implementations fo the diff methods to see whether it uses the compareTo call the same way that the SafetyUtil.equals method does. If it is more correct in more cases, then we should switch it over .. but I'm not sure whether that is true. If you have examples of the implementation of the diff() method in java for different property types, send them over ...